src/openms/source/ANALYSIS/QUANTITATION/IsotopeLabelingMDVs.cpp

```diff
@@ -0,0 +1,322 @@
+      {
+        std::vector<OpenMS::Peak2D::IntensityType> intensities_vec;
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          intensities_vec.push_back(it->getMetaValue(NamesOfFeatureName[static_cast<int>(FeatureName::PEAK_APEX_INT)]));
+        }
+        std::vector<OpenMS::Peak2D::IntensityType>::iterator max_it = std::max_element(intensities_vec.begin(), intensities_vec.end());
+        double measured_feature_max = intensities_vec[std::distance(intensities_vec.begin(), max_it)];
+          
+        if (measured_feature_max != 0.0)
+        {
+          for (size_t i = 0; i < normalized_feature.getSubordinates().size(); ++i)
+          {
+            normalized_feature.getSubordinates().at(i).setIntensity((OpenMS::Peak2D::IntensityType)measured_feature_subordinates.at(i).getMetaValue(NamesOfFeatureName[static_cast<int>(FeatureName::PEAK_APEX_INT)]) / measured_feature_max);
+          }
+        }
+      }
+    }
+    else if (mass_intensity_type == MassIntensityType::NORM_SUM)
+    {
+      if (feature_name == FeatureName::INTENSITY)
+      {
+        OpenMS::Peak2D::IntensityType feature_peak_apex_intensity_sum = 0.0;
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          feature_peak_apex_intensity_sum += it->getIntensity();
+        }
+        
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          normalized_feature.setMetaValue((it - measured_feature_subordinates.begin()), (it->getIntensity() / feature_peak_apex_intensity_sum));
+        }
+      }
+      // for every other case where feature_name isn't 'intensity', i.e. 'peak_apex_int'
+      else if (feature_name == FeatureName::PEAK_APEX_INT)
+      {
+        OpenMS::Peak2D::IntensityType feature_peak_apex_intensity_sum = 0.0;
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          feature_peak_apex_intensity_sum += (Peak2D::IntensityType)it->getMetaValue(NamesOfFeatureName[static_cast<int>(FeatureName::PEAK_APEX_INT)]);
+        }
+
+        if (feature_peak_apex_intensity_sum != 0.0)
+        {
+          for (size_t i = 0; i < normalized_feature.getSubordinates().size(); ++i)
+          {
+            normalized_feature.getSubordinates().at(i).setIntensity((OpenMS::Peak2D::IntensityType)measured_feature_subordinates.at(i).getMetaValue(NamesOfFeatureName[static_cast<int>(FeatureName::PEAK_APEX_INT)]) / feature_peak_apex_intensity_sum);
+          }
+        }
+      }
+    }
+  }
+
+  void IsotopeLabelingMDVs::calculateMDVs(
+    const FeatureMap& measured_featureMap, FeatureMap& normalized_featureMap,
+    const MassIntensityType& mass_intensity_type, const FeatureName& feature_name)
+  {
+    if (!normalized_featureMap.empty())
```

Comment:
no need for the surrounding `if ...`?! Just always clear..
It's less code and faster to parse for a human